metadata-db-app/src/cmr/metadata_db/data/memory_db.clj

```diff
@@ -131,6 +132,111 @@
        :existing-concept-id existing-concept-id
        :existing-native-id existing-native-id})))
 
+(defn- concept-id-not-in-list?
```

Comment:
nit: I'm not sure having two functions for `concept-id-in-list` is really necessary when `not` works well but in any case the logic of `concept-id-in-list?` should be contained in one function. So `concept-id-not-in-list?` should just call `(not (concept-id-in-list? list concept-id))`

Reply:
I thought about the same thing. but the logic is so simple that I thought it doesn't really matter.

Reply:
Removed one function. I thought it's more readable this way, but it's readable the other way too.